packages/amplify-opensearch-simulator/src/__tests__/index.test.ts

```diff
@@ -0,0 +1,169 @@
+  const startupErrorMessage = 'Unable to start the Opensearch emulator. Please restart the mock process.';
+
+  const openSearchClusterOptions = {
+    clusterName: 'mock-opensearch-cluster',
+    nodeName: 'mock-opensearch-node-local',
+    port: 9600,
+    type: 'single-node',
+    version: '1.3.0'
+  };
+
+  const openSearchClusterDefaultOptions = {
+    clusterName: 'opensearch-cluster',
+    nodeName: 'opensearch-node-local',
+    port: 9200,
+    type: 'single-node',
+    version: '1.3.0'
+  };
+
+  const ensureMockSearchableResourcePath = () => {
+    fs.ensureDirSync(mockSearchableResourcePath);
+    fs.emptyDirSync(mockSearchableResourcePath);
+  };
+
+  let emulators: openSearchEmulator.OpenSearchEmulator[];
+  beforeEach(async () => {
+    ensureMockSearchableResourcePath();
+    emulators = [];
+    jest.setTimeout(5 * 60 * 1000);
+    jest.resetModules();
+    jest.spyOn(openpgp, 'verify').mockReturnValueOnce({
+      signatures: [{
+        verified: Promise.resolve(true)
+      }]
+    } as $TSAny);
+  });
+
+  afterEach(async () => {
+    await Promise.all(emulators.map(emu => emu.terminate()));
+  });
+
+  afterAll(async () => {
+    ensureMockSearchableResourcePath();
+  });
+
+  const fetchURL = async (url: string): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      http
+        .get(url, (resp) => {
+          let data = "";
+          resp.on("data", (chunk) => {
+            data += chunk;
+          });
+          resp.on("end", () => {
+            resolve(data);
+          });
+        })
+        .on("error", (err) => {
+          reject(err);
+        });
+    });
+  };
+
+  if (isWindowsPlatform()) {
+    it('should fail to launch on windows OS', async () => {
+      try {
+        await openSearchEmulator.launch(mockSearchableResourcePath);
+        fail('launching simulator is expected to throw but did not');
+      } catch (error) {
+        expect(error.message).toEqual('Cannot launch OpenSearch Simulator on windows OS');
+      }
+    });
+  } else {
+    it('correctly resolves the path to local opensearch binary', async () => {
+      const relativePathFromMockSearchableResourceDir = await openSearchEmulator.getPathToOpenSearchBinary();
+      expect(relativePathFromMockSearchableResourceDir).toEqual(join('opensearchLib', 'bin', 'opensearch'));
+      
+      const fullPathToOpenSearchBinary = await openSearchEmulator.getPathToOpenSearchBinary(mockSearchableResourcePath);
+      expect(fullPathToOpenSearchBinary).toEqual(join(mockSearchableResourcePath, 'opensearchLib', 'bin', 'opensearch'));
+    });
+    
+    it('skips downloading another opensearch binary when one is locally available', async () => {
+      const openSearchExists = await openSearchEmulator.openSearchLocalExists(mockSearchableResourcePath);
+      // returns false when there is no local binary
+      expect(openSearchExists).toEqual(false);
+
+      const nodefetch = require('node-fetch');
```

Comment:
Why dynamically require node-fetch if mocking it on the next line?

Reply:
I want to check the number of calls to this method which requires me to mock it. I however, only want to mock it locally for this test, hence the dynamic import.

Reply:
```suggestion
      const nodefetch = await import('node-fetch');
```

Reply:
Please use camel-case for variable names. The name in this case should be nodeFetch.